spec/features/two_factor_authentication/sign_in_spec.rb

```diff
@@ -489,15 +489,28 @@ def attempt_to_bypass_2fa
     end
   end
 
-  describe 'second factor locked' do
+  describe 'rate limiting' do
+    let(:max_attempts) { 2 }
+    let(:user) { create(:user, :signed_up) }
+
     before do
-      allow_any_instance_of(UserDecorator).to receive(:locked_out?).and_return(true)
-      allow_any_instance_of(UserDecorator).to receive(:lockout_time_expiration).
-        and_return(Time.zone.now + 10.minutes)
+      allow(IdentityConfig.store).to receive(:login_otp_confirmation_max_attempts).
+        and_return(max_attempts)
+    end
+
+    def wrong_phone_otp
+      loop do
+        code = rand(100000...999999).to_s
+        return code if code != last_phone_otp
+      end
     end
```

Comment:
May be overkill; in other places we just use a hard-coded value ([example](https://github.com/18F/identity-idp/blob/d44cac20c8aa65620c3f9ee1ea810e9edcdc8362/spec/features/two_factor_authentication/change_factor_spec.rb#L83)), but a spec which flakes 1 in 900,000 times is still a flakey spec. 🤷

Reply:
To be fair, that other example is guaranteed to fail because it's 5 digits (instead of 6), but it's planned to add some client-side validation for code length as part of [LG-8067](https://cm-jira.usa.gov/browse/LG-8067). That'd still require ChromeDriver to be active to be enforced, but this would be more compatible if ever we moved in the direction of defaulting to ChromeDriver for feature specs.